pkg/apis/azure/v1alpha1/types_worker.go

```diff
@@ -49,10 +54,25 @@ type MachineImage struct {
 	Name string `json:"name"`
 	// Version is the logical version of the machine image.
 	Version string `json:"version"`
+	// AcceleratedNetworking is an indicator if the image supports Azure accelerated networking.
+	// +optional
+	AcceleratedNetworking *bool `json:"acceleratedNetworking,omitempty"`
+	// Architecture is the CPU architecture of the machine image.
+	// +optional
+	Architecture *string `json:"architecture,omitempty"`
+	// SkipMarketplaceAgreement skips the marketplace agreement check when enabled.
+	// +optional
+	SkipMarketplaceAgreement *bool `json:"skipMarketplaceAgreement,omitempty"`
+	// Image identifies the azure image.
+	Image
```

Comment:
I am not sure if we want to do that or something like:

```
imageRef Image `json",inline"`
```

I think this is more clean to be honest and I believe this to be backwards compatible too. WDYT ?